### PR TITLE
Refactor stable API to not use `client._req`

### PR DESCRIPTION
**Pull request by bbonenfant**
Decided to split the task into refactoring stable and refactoring the experimental API because it's diverged quite a bit. This is also the reason why I'm making a new PR as opposed to continuing #383 

JIRA issues: [INT-492]

[INT-492]: https://pachyderm.atlassian.net/browse/INT-492?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/python_pachyderm/mixin/auth.py [hunk 1]
```diff
@@ -177,7 +186,8 @@ def authenticate_oidc(self, oidc_state: str) -> str:
         str
             A token that can be used for making authenticate requests.
         """
-        return self._req(Service.AUTH, "Authenticate", oidc_state=oidc_state).pach_token
+        message = auth_pb2.AuthenticateRequest(oidc_state=oidc_state)
+        return self.__stub.Authorize(message).pach_token
```
Comment (smalyala) [inline src/python_pachyderm/mixin/auth.py:190]: This should be `self.__stub.Authenticate()`. Might be good to get coverage on these methods in the tests.

File: src/python_pachyderm/mixin/auth.py [hunk 2]
```diff
@@ -194,13 +204,14 @@ def authenticate_id_token(self, id_token: str) -> str:
         str
             A token that can be used for making authenticate requests.
         """
-        return self._req(Service.AUTH, "Authenticate", id_token=id_token).pach_token
+        message = auth_pb2.AuthenticateRequest(id_token=id_token)
+        return self.__stub.Authorize(message).pach_token
```
Comment (smalyala) [inline src/python_pachyderm/mixin/auth.py:208]: This should be `self.__stub.Authenticate()`.

File: src/python_pachyderm/mixin/debug.py [hunk 3]
```diff
@@ -1,13 +1,22 @@
...
+from python_pachyderm.proto.v2.debug import debug_pb2, debug_pb2_grpc
+
 
 class DebugMixin:
     """A mixin for debug-related functionality."""
 
+    _channel: grpc.Channel
+
+    def __init__(self):
+        self.__stub = debug_pb2_grpc.DebugStub(self._channel)
```
Comment (smalyala) [inline src/python_pachyderm/mixin/debug.py:15]: It's funny to me that proto decides to spit out `DebugStub` rather than `ApiStub` here.

**Comment by smalyala** [comment]
Check out `util.py`. That has some `pps_proto` references.

**Comment by smalyala** [comment]
What are your thoughts on also changing module names in the docstrings (i.e. "Returns `pfs_proto.Repo` -> `pb2_proto.Repo`")?

File: src/python_pachyderm/mixin/pfs.py [hunk 4]
```diff
@@ -1047,18 +1038,18 @@ def list_file(
         --------
         >>> files = list(client.list_file(("foo", "master"), "/dir/subdir/"))
         """
-        return self._req(
-            Service.PFS,
-            "ListFile",
-            file=pfs_proto.File(commit=commit_from(commit), path=path, datum=datum),
+        message = pfs_pb2.ListFileRequest(
+            details=details,
```
Comment (smalyala) [inline src/python_pachyderm/mixin/pfs.py:1042]: No harm in adding it back in but fyi `details` isn't actually used

File: src/python_pachyderm/mixin/auth.py (hunk 1 above)
Comment (bbonenfant) [reply src/python_pachyderm/mixin/auth.py:190]: Good catch, I'd definitely like to increase the coverage here, but I'm not exactly sure of a great way to test it. any ideas?

**Review by smalyala — APPROVED** [review]
Sorry, lost track of this PR. LGTM!

**Pull request merged**